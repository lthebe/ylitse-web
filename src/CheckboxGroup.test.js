import React from 'react';
import { shallow, mount } from 'enzyme';
import CheckboxGroup from './CheckboxGroup';

describe('CheckboxGroup', () => {
    const mockFunc = jest.fn();
    const languages = { Finnish: false };
    const props = {
        data: languages,
        className: '',
        label: 'Language',
        onChange: mockFunc,
    };

    test('renders correctly', () => {
        const wrapper = shallow(<CheckboxGroup {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('checkbox can be checked and unchecked', () => {
        const wrapper = mount(<CheckboxGroup {...props} />);
        const input = wrapper.find('input[name="Finnish"]');
        input.simulate('change');
        expect(mockFunc).toHaveBeenCalled();
    });
});
