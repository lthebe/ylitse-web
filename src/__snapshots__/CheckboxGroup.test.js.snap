// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckboxGroup
    className=""
    data={
      Object {
        "Finnish": false,
      }
    }
    label="Language"
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(FormLabel)
          component="legend"
        >
          Language
        </WithStyles(FormLabel)>,
        <WithStyles(FormGroup)
          row={true}
        >
          <WithStyles(FormControlLabel)
            checked={false}
            control={
              <WithStyles(Checkbox)
                onChange={[MockFunction]}
              />
            }
            label="Finnish"
            name="Finnish"
          />
        </WithStyles(FormGroup)>,
      ],
      "className": "",
      "component": "fieldset",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Language",
          "component": "legend",
        },
        "ref": null,
        "rendered": "Language",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(FormControlLabel)
              checked={false}
              control={
                <WithStyles(Checkbox)
                  onChange={[MockFunction]}
                />
              }
              label="Finnish"
              name="Finnish"
            />,
          ],
          "row": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Finnish",
            "nodeType": "class",
            "props": Object {
              "checked": false,
              "control": <WithStyles(Checkbox)
                onChange={[MockFunction]}
              />,
              "label": "Finnish",
              "name": "Finnish",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(FormLabel)
            component="legend"
          >
            Language
          </WithStyles(FormLabel)>,
          <WithStyles(FormGroup)
            row={true}
          >
            <WithStyles(FormControlLabel)
              checked={false}
              control={
                <WithStyles(Checkbox)
                  onChange={[MockFunction]}
                />
              }
              label="Finnish"
              name="Finnish"
            />
          </WithStyles(FormGroup)>,
        ],
        "className": "",
        "component": "fieldset",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Language",
            "component": "legend",
          },
          "ref": null,
          "rendered": "Language",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(FormControlLabel)
                checked={false}
                control={
                  <WithStyles(Checkbox)
                    onChange={[MockFunction]}
                  />
                }
                label="Finnish"
                name="Finnish"
              />,
            ],
            "row": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Finnish",
              "nodeType": "class",
              "props": Object {
                "checked": false,
                "control": <WithStyles(Checkbox)
                  onChange={[MockFunction]}
                />,
                "label": "Finnish",
                "name": "Finnish",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
